upgrade autotools for aarch64 depends build

ghost opened the pull request:
upgrade autotools for aarch64 depends build
## Description

upgrade autotools for aarch64 depends build
config.sub/guess are the latest ones from autotools side
## Motivation and Context

upgrade autotools for aarch64 depends build
## How Has This Been Tested?

cross compiled on ubuntu x86-64 14.04 lts
## Types of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

koying committed:
CHG: [NDK21] sambav3

Force all off_t to off64_t

Commit:
upgrade autotools for aarch64 depends build

wsnipex commented on tools/depends/configure.ac:
```diff
@@ -169,6 +169,18 @@ case $host in
+        use_cpu=$host_cpu
+      fi
+      use_toolchain="${use_toolchain:-/usr}"
+      platform_cflags="-Os -fPIC -DPIC"
```
are we sure we want -Os here? @stefansaraev any opinion?

stefansaraev replied:
it's a c/p from arm*-*-linux-gnu few lines above and it does not hurt.

wsnipex reviewed (commented):
thanks!
a bit hard to review due to the config.sub/guess update, but looks good to me.

wsnipex commented:
jenkins build this please

Memphiz commented:
mhh depends was already able to build with aarch64 / arm64 for ios/tvos. I have to check if this is still working with this PR - but can't do it until sunday.

wsnipex committed:
[cmake] ffmpeg: add android aarch64 support

wsnipex committed:
[android] add aarch64 support

Commit:
aarch64 compile stuff

Memphiz commented:
tvos and ios arm64 depends still building fine with this ... good from me.

mapfau commented:
still not 100% finished.... final commit will arrive soon

wsnipex committed:
[cmake] android: resolve symlinks when copying libs to install dir

wsnipex committed:
[depends] set CMAKE_STRIP in Toolchain.cmake

wsnipex committed:
[android] fix packaging for aarch64

Commit:
bump default sdk version to 21 / remove offs64 overrides in dvdnav/read

wsnipex commented:
jenkins build this please

Memphiz commented:
That last commit sounds like it would need a change for Jenkins too here no?:

https://github.com/xbmc/xbmc/blob/master/tools/buildsteps/defaultenv#L33

IIRC API 21 is installed on the slaves already

wsnipex committed:
[cmake] Macros: use ${CMAKE_COMMAND} instead of "cmake"

wsnipex committed:
[android] allow building with jdk 8

TODO: use cmake's Java_JAVAC_EXECUTABLE var and version instead of calling javac directly

Commit:
[Android] Fix BundleFiles RegExp

hudokkow commented:
jenkins build this please

mapfau commented:
1.) stripping still fails from time to time (make apk)

2.) Some addons fail with the droid::32bit build:
Jenkins: http://jenkins.kodi.tv/job/Android-ARM/9361/

@wsnipex @garbear here is one example:

"/home/jenkins/workspace/android-arm/tools/depends/target/binary-addons/arm-linux-androideabi-android-21/build/pcre/src/pcre/pcrecpp.cc:855:50: error: 'strtouq' was not declared in this scope
  unsigned long long r = strtouq(str, &end, radix);

Any idea (strouq????, whats this?)

wsnipex commented:
@Memphiz yes, this needs ndk 12. I tried with ndk 10, but it fails on aarch64.

@mapfau some addons always fail, we won't be able to fix this in this PR. As long as not _all_ fail, I wouldn't care atm.

mapfau commented:
most of the addons are running well.
some fail because of header files (sys/timeb.h), some because of reasons as above.

Commit:
version bump / strip force

Commit:
add aarch64 buildstep

Commit:
rename builstep folder freom aarch -> arm64

Commit:
packaging fix

mapfau commented:
will fix soon :-)

peak3d committed:
remove android-aarch64 / splash fix for arm64

wsnipex commented:
as said on slack, you have to rebase project/cmake/modules/FindLibDvd.cmake

peak3d committed:
pass CPU instead ARCH as xbmc.property

Commit:
FIX: [aarch64] elf64 support. Credits to koying

wsnipex committed:
[cmake] set DT_SONAME in wrapped libs

Commit:
FIX: [droid] don't try /proc/self/exe (selinux). credits to koying

wsnipex committed:
[cmake] android: link wrap libs against libdummy

fetzerch commented on xbmc/cores/DllLoader/exports/CMakeLists.txt:
```diff
@@ -21,7 +21,7 @@ elseif(NOT CORE_SYSTEM_NAME STREQUAL windows)
   if(CORE_SYSTEM_NAME STREQUAL android)
-    add_custom_command(TARGET wrapper.def COMMAND echo \"-l$<TARGET_FILE:${APP_NAME_LC}>\" >> wrapper.def)
+    add_custom_command(TARGET wrapper.def COMMAND echo \"-L${kodi_BINARY_DIR} -l:libkodi.so\" >> wrapper.def)
     add_dependencies(wrapper.def ${APP_NAME_LC})
```
You could use the TARGET_FILE_DIR and TARGET_FILE_NAME generator expression. (https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html)
They have the advantage that cmake already complains if the library target does for some reason not exist. So it makes things more robust. `kodi` should always be APP_NAME_LC, otherwise it breaks builds where someone tries to rebrand.  Also I'm wondering where the kodi_BINARY_DIR comes from.

wsnipex replied:
this is wrong anyway, it is -l[name] without lib prefix and .so suffix.
 i.e. -lkodi
Anyway, autotools links against libdummy-kodi, so see https://github.com/wsnipex/xbmc/commit/07477c1dbd7a6f953636a8e2ba7089cbca3c6134

mapfau replied:
its fom an android documentation with ":", seems to work well.
But, why use dummy? The problem seems more that the full path is given in -l

mapfau replied:
Pls let me know what way you prefer. I'll make the final shot then

wsnipex replied:
I prefer it the same as in autotools unless someone can explain why libdummy-kodi is not needed.
You can simply reset on my branch, I also fixed an rebase error there.

fetzerch commented on xbmc/platform/android/jni/AudioFormat.h:
```diff
@@ -19,6 +19,8 @@
  *
  */
 
+#define HAVE_INTTYPES_H 1
```
are these needed when you set the define from cmake/autotools?

mapfau replied:
@wsnipex I agree, they shouldn't be in source files but in cmake project

wsnipex replied:
see top 2 commits: https://github.com/wsnipex/xbmc/commits/aarch64

warning: this is _not_ build tested

Commit:
ADD: [droid] capture stderr/stdout to logcat / thanks to @koying

wsnipex committed:
[cmake] check for inttypes

wsnipex committed:
don't define INTTYPES_H in core

Commit:
add aarch64 lock-obj patch to libgpg-error

peak3d commented:
moved to https://github.com/xbmc/xbmc/pull/10659
